MultiThreading/thread_example.cpp

```diff
@@ -2,16 +2,25 @@
 #include <pthread.h>
 #include <unistd.h>
 #include <queue> // FIFO, pilas son LIFO
+#include <semaphore.h>
 
 #define MAX_QUEUE_SIZE    10
 // producer-consumer problem
 // race conditions
 // dos threads accediendo y/o modificando un recurso compartido int flag, queue, list 
 
 std::queue<int> myQueue;
+sem_t semaphore_full;
```

Comment:
sem_t is a struct or a class type? @ramsesrago

Reply:
yes,. it's a struct